File: 01_Array/Q17_13. Roman to Integer/solution.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_map = {
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000
        }

        integer_value = 0
        for i in range(len(s)):
            if i+1 < len(s) and roman_map[s[i]] < roman_map[s[i + 1]]:
                integer_value -= roman_map[s[i]]
            else:
                integer_value += roman_map[s[i]]
                
        return integer_value


# Time Complexity -> O(n)
# Space Complexity -> O(n)

File: 01_Array/Q17_13. Roman to Integer/solution.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> mpp;
        int n = s.length();
        int intergerVal = 0;

        mpp['I'] = 1;
        mpp['V'] = 5;
        mpp['X'] = 10;
        mpp['L'] = 50;
        mpp['C'] = 100;
        mpp['D'] = 500;
        mpp['M'] = 1000;

        for(int i = 0; i < n; i++){
            if(mpp[s[i]] < mpp[s[i+1]]) 
                intergerVal -= mpp[s[i]];
            else
                intergerVal += mpp[s[i]];
        }

        return intergerVal;
    }
};



// Time Complexity -> O(n)
// Space Complexity -> O(n)
